src/control/modify.js

```diff
@@ -46,17 +46,39 @@ const selectModifyStyle = new Style({
   }),
   geometry: (f) => {
     let coordinates = [];
-    if (f.getGeometry().getType() === 'Polygon') {
-      f.getGeometry()
-        .getCoordinates()[0]
-        .forEach((coordinate) => {
-          coordinates.push(coordinate);
-        });
-    } else if (f.getGeometry().getType() === 'LineString') {
-      coordinates = f.getGeometry().getCoordinates();
-    } else {
-      coordinates = [f.getGeometry().getCoordinates()];
+    const geometry = f.getGeometry();
+    let geometries = [geometry];
+    if (geometry.getGeometriesArrayRecursive) {
+      geometries = geometry.getGeometriesArrayRecursive();
     }
+    
+    // At this point geometries doesn't contains any GeometryCollections.
+    geometries.forEach((geom)=>{
+      const multiGeometries = [geom];
+      if (geom.getLineStrings) {
```

Comment:
If the api changes I also have to change the next line . That's why I test if the function exist instead of comparing the type. If they change the type string for some reason, everything is broken.

Reply:
Finally I've used th real string from ol , it should more robust